##SOAP
```
/home/evanslab/tetra_project/Assembly/SOAP/SOAPdenovo2-src-r240/SOAPdenovo-127mer all -s /net/infofile2/2/scratch/evanslab_backups/2014_BJE3652_gDNA_seqs/Mellotropicalis.conf -K 65 -R -N 3400000000 -p 10 -V -o /net/infofile2/2/scratch/evanslab_backups/2014_BJE3652_gDNA_seqs/SOAP_Mellotropicalis_BJE3652_genome

- Error: Segmentation fault when trying to import Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence_09082015_NoIndex_L008_R2_014_trim_paired.fastq.gz

/home/evanslab/tetra_project/Assembly/SOAP/SOAPdenovo2-src-r240/SOAPdenovo-127mer all -s Mellotropicalis.conf -K 65 -R -N 3400000000 -p 10 -V -o SOAP_Mellotropicalis_BJE3652_genome 1>ass.log 2>ass.err

- Error: Ambiguous output redirect

SOAPdenovo-127mer all -s Mellotropicalis_without_180_2ndSeq.conf -K 65 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome

After: try increasing Kmer size
SOAPdenovo-127mer all -s Mellotropicalis.conf -K 80 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome

After: try decreasing Kmer size  
SOAPdenovo-63mer all -s Mellotropicalis.conf -K 63 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome
```
Working without 180_2ndSeq but results not that good (small scaffolds): try with bigger kmer size (possible because we have a lot of data)
```
SOAPdenovo-127mer all -s Mellotropicalis_without_180_2ndSeq_125mers.conf -K 125 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_125

SOAPdenovo-127mer all -s Mellotropicalis_without_180_2ndSeq_125mers.conf -K 80 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_80
Segmentation fault
```
Multi k-mer approach
```
SOAPdenovo-127mer all -s Mellotropicalis_without_180_2ndSeq_125mers.conf -K 65 -m 125 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome65_to_125
(killed)

SOAPdenovo-127mer all -s Mellotropicalis_without_180_2ndSeq_125mers.conf -K 65 -m 95 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_65_to_95

SOAPdenovo-63mer all -s Mellotropicalis_without_180_2ndSeq_125mers.conf -K 43 -m 63 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_43_to_63
```
Only with 180 library
```
SOAPdenovo-127mer all -s Mellotropicalis_180_only.conf -K 65 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_180_only_65mers
Import reads from file:
 /net/infofile2/2/scratch/evanslab_backups/2014_BJE3652_gDNA_seqs/Sample_Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence_09082015_untar/Sample_Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence_09082015/only_trim_paired_AND_Half/Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence_09082015_NoIndex_L006_R2_041_trim_paired.fastq.gz
Segmentation fault

SOAPdenovo-63mer all -s Mellotropicalis_180_L006_only.conf -K 61 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_180_L006_only_61mers
```
Genome size:

*[X. tropicalis](http://www.xenbase.org/anatomy/intro.do)*: 1.7 x 10^9 bp x 2 (duplicated genome) = 3.4 x 10^9

###Try to improve
If we obtain an OK assembly, we can try some "diploid" assembly program to try improving it.
- [ScaffoldScaffolder](http://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-16-S7-S5) seems promissing for dealing with diploid assembly using bubbles to identify heterozygous homologous. Should be good for sites when each subgenome is homozygous but how to deal with heteroz. sites in both/one of the sub-genomes?  
- Haplotype assembly using *X. tropicalis* as the ref genome?
 - SDhaP  
 - http://www.ncbi.nlm.nih.gov/pubmed/16077012


- Try [Platanus](http://platanus.bio.titech.ac.jp/platanus/?page_id=14) that pretends to deal with highly heterozygous? genomes
