## Reference
Structure: `REFERENCE/DATA/RUN/ASSEMBLIES/SUBDIR`.

In the ALLPATHS it is advice to use a reference genome if one is available to help building the genome: using *X.tropicalis*.
DATA directories = sets of reads to assemble.
Arguments to give 

> PRE – the root directory in which the ALLPATHS pipeline directory will be created. (= '/work/ben/Mellotropicalis_corrected_data')

> REFERENCE_NAME – the REFERENCE (organism) directory name 

> DATA_SUBDIR – the DATA (project) directory name

> RUN – the RUN (assembly pre ‐ processing) directory name (='allpaths')

> SUBDIR – the SUBDIR (assembly) directory name

#### Preparing the data
We have paired-end libraries. We need to rename the files `Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence_09082015_NoIndex_L008_R1_011_trim_paired.fastq.gz` needs to become `Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence_09082015_NoIndex_L008_R1_011_trim_paired.A.fastq.gz` .
```
(on scratch)
module load allpathslg
PrepareAllPathsInputs.pl DATA_DIR=/scratch/ben/allpath_assembly/X_mellotropicalis/data PLOIDY=1 IN_GROUPS_CSV=/scratch/ben/allpath_assembly/X_mellotropicalis/in_groups.csv IN_LIBS_CSV=/scratch/ben/allpath_assembly/X_mellotropicalis/in_libs.csv 
```
`PLOIDY` only 0 (haploid) or 1 (diploid) no other ploidy level tested.
The `DATA_DIR` needs to contain the `in_groups.csv` & `in_libs.csv` files.
The reference genome must be supplied as two files: genome.fasta and genome.fastb. (Run `Fasta2Fastb`) - 1st we will do it without any ref. seq (`Run1`).

`in_groups.csv`
```
group_name,library_name,file_name
6kb,Illumina_6kb,/scratch/ben/allpath_assembly/X_mellotropicalis/data/Ben-Evans-BJE3652-6kb_NoIndex_*.fastq.gz
1kb,Illumina_1kb,/scratch/ben/allpath_assembly/X_mellotropicalis/data/BenEvansBJE3652_1000bp_Library*.fastq.gz
10kb,Illumina_10kb,/scratch/ben/allpath_assembly/X_mellotropicalis/data/Ben_Evans_BJE3652_10kb_Mate_Pair_Library*.fastq.gz
180bp_2,Illumina_180bp_2,/scratch/ben/allpath_assembly/X_mellotropicalis/data/Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence*.fastq.gz
180bp_1,Illumina_180bp,/scratch/ben/allpath_assembly/X_mellotropicalis/data/BenEvansBJE3652_180bp_Library_GTTTCG*.fastq.gz
400bp,Illumina_400bp,/scratch/ben/allpath_assembly/X_mellotropicalis/data/BenEvansBJE3652_400bp*.fastq.gz
6kb_2,Illumina_6kb_2,/scratch/ben/allpath_assembly/X_mellotropicalis/data/Ben_Evans_BJE3652_6kb_2nd_Sequencing*.fastq.gz
```
`in_libs.csv`
```
library_name,project_name,organism_name,type,paired,frag_size,frag_stddev,insert_size,insert_stddev,read_orientation,genomic_start,genomic_end
Illumina_6kb,Mellotropicalis_poject,Xenopus_mellotropicalis,jumping,1,,,6000,500,inward,0,0
Illumina_1kb,Mellotropicalis_poject,Xenopus_mellotropicalis,fragment,1,90,60,,,inward,0,0
Illumina_10kb,Mellotropicalis_poject,Xenopus_mellotropicalis,jumping,1,,,10000,500,inward,0,0
Illumina_180bp,Mellotropicalis_poject,Xenopus_mellotropicalis,fragment,1,90,60,,,inward,0,0
Illumina_180bp_2,Mellotropicalis_poject,Xenopus_mellotropicalis,fragment,1,90,60,,,inward,0,0
Illumina_400bp,Mellotropicalis_poject,Xenopus_mellotropicalis,fragment,1,90,60,,,inward,0,0
Illumina_6kb_2,Mellotropicalis_poject,Xenopus_mellotropicalis,jumping,1,,,6000,500,inward,0,0
```
```
RunAllPathsLG
PRE=/scratch/ben/allpath_assembly
REFERENCE_NAME=X_mellotropicalis
DATA_SUBDIR=data
RUN=Run1
TARGETS=standard
EVALUATION=standard
THREADS=4
```
!!!! Differences with Abyss & SOAPdenovo: no need of specify a k-mer size (default `K=96`). No direct linked between k-mer size and read length. Allpaths uses multiple values when building the assembly. 

##On test data
```
module load allpathslg
PrepareAllPathsInputs.pl DATA_DIR=/scratch/ben/allpath_assembly/ALLPATHS-LG.test_genome/seq IN_GROUPS_CSV=/scratch/ben/allpath_assembly/ALLPATHS-LG.test_genome/seq/in_groups.csv IN_LIBS_CSV=/scratch/ben/allpath_assembly/ALLPATHS-LG.test_genome/seq/in_libs.csv
RunAllPathsLG PRE=/scratch/ben/allpath_assembly/ REFERENCE_NAME=ALLPATHS-LG.test_genome DATA_SUBDIR=seq RUN=myrun TARGETS=standard 
```
(`myrun` is created while running, needs a `ploidy` file in the DATA directory)
