##SOAP
```
/home/evanslab/tetra_project/Assembly/SOAP/SOAPdenovo2-src-r240/SOAPdenovo-127mer all -s /net/infofile2/2/scratch/evanslab_backups/2014_BJE3652_gDNA_seqs/Mellotropicalis.conf -K 65 -R -N 3400000000 -p 10 -V -o /net/infofile2/2/scratch/evanslab_backups/2014_BJE3652_gDNA_seqs/SOAP_Mellotropicalis_BJE3652_genome

- Error: Segmentation fault when trying to import Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence_09082015_NoIndex_L008_R2_014_trim_paired.fastq.gz

/home/evanslab/tetra_project/Assembly/SOAP/SOAPdenovo2-src-r240/SOAPdenovo-127mer all -s Mellotropicalis.conf -K 65 -R -N 3400000000 -p 10 -V -o SOAP_Mellotropicalis_BJE3652_genome 1>ass.log 2>ass.err

- Error: Ambiguous output redirect

SOAPdenovo-127mer all -s Mellotropicalis_without_180_2ndSeq.conf -K 65 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome


After: try increasing Kmer size
SOAPdenovo-127mer all -s Mellotropicalis.conf -K 80 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome

After: try decreasing Kmer size  
SOAPdenovo-63mer all -s Mellotropicalis.conf -K 63 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome
```
Results `SOAP_Mellotropicalis_BJE3652_genome` (65mers):
```
Size_includeN	5076388490
Size_withoutN	2697808415
Scaffold_Num	5507366
Mean_Size	921
Median_Size	131
Longest_Seq	148529
Shortest_Seq	100
Singleton_Num	4852830
Average_length_of_break(N)_in_scaffold	431

Known_genome_size	-794967296
Total_scaffold_length_as_percentage_of_known_genome_size	-638.57%

scaffolds>100 	5437123	98.72%
scaffolds>500 	820199	14.89%
scaffolds>1K  	494127	8.97%
scaffolds>10K 	204740	3.72%
scaffolds>100K	30	0.00%
scaffolds>1M  	0	0.00%

Nucleotide_A	820282695	16.16%
Nucleotide_C	538445828	10.61%
Nucleotide_G	530414447	10.45%
Nucleotide_T	808665445	15.93%
GapContent_N	2378580075	46.86%
Non_ACGTN	0	0.00%
GC_Content	39.62%		(G+C)/(A+C+G+T)

N10	33138	11591
N20	23425	29748
N30	19817	53560
N40	13214	85129
N50	11244	127377
N60	10358	174568
N70	6499	229794
N80	1079	467635
N90	206	1666368

NG50	148529	1
N50_scaffold-NG50_scaffold_length_difference	137285
```
Working without 180_2ndSeq but results not that good (small scaffolds): try with bigger kmer size (possible because we have a lot of data)
```
SOAPdenovo-127mer all -s Mellotropicalis_without_180_2ndSeq_125mers.conf -K 125 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_125

SOAPdenovo-127mer all -s Mellotropicalis_without_180_2ndSeq_125mers.conf -K 80 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_80
Segmentation fault
```
On iqaluk
```
/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-63mer all -s Mellotropicalis_all_libraries.conf -K 43 -m 63 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_all_libraries_43_to_63
Stuck forever in one step 

/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-127mer all -s Mellotropicalis_all_libraries.conf -K 65 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_all_libraries_65
Bus error
/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-127mer all -s Mellotropicalis_all_libraries.conf -K 67 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_all_libraries_67
/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-63mer all -s Mellotropicalis_all_libraries.conf -K 63 -R -F  -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_all_libraries_63
```

**Also need to try including big library (>1kb) for only scaffolding**

**Note 09/08:** 
- Multi kmer approach use too much time & memory for big dataset. Will use the kmer size with which we have obtained the best results when we didn't consider the 2nd sequencing of 180bp library `-K 65`. And also `-K 67` (considering read length is around 100bp, `2/3 * 10 ~ 66.667` so should obtain best assembly with kmer size around that value in theory. Trying smaller value `43-63` and bigger `65-95` didn't improve the assembly (when considering the data without 2nd Sequencing og 180bp library). 
- Should use [velvet](https://github.com/dzerbino/velvet/blob/master/doc/manual_src/Manual.tex) and merge multiple assemblies obtained with different kmer size (65 & 67 at least) to compare the result. [Allpaths](http://software.broadinstitute.org/allpaths-lg/blog/?page_id=12) seems to be a good alternative option too for our dataset (but not for *Hymenochirus*):

***ALLPATHSâ€LG requires high sequence coverage of the genome in order to compensate for the shortness of the reads. The precise coverage required depends on the length and quality of the paired reads, but typically is of the order 100x or above.***

Multi k-mer approach
```
SOAPdenovo-127mer all -s Mellotropicalis_without_180_2ndSeq_125mers.conf -K 65 -m 125 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome65_to_125
(killed)

SOAPdenovo-127mer all -s Mellotropicalis_without_180_2ndSeq_125mers.conf -K 65 -m 95 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_65_to_95

SOAPdenovo-63mer all -s Mellotropicalis_without_180_2ndSeq_125mers.conf -K 43 -m 63 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_43_to_63
```
Result `SOAP_Mellotropicalis_BJE3652_genome_43_to_63`:
```
Size_includeN	6150731839
Size_withoutN	3155866842
Scaffold_Num	8993155
Mean_Size	683
Median_Size	127
Longest_Seq	113293
Shortest_Seq	100
Singleton_Num	8172338
Average_length_of_break(N)_in_scaffold	333

Known_genome_size	-794967296
Total_scaffold_length_as_percentage_of_known_genome_size	-773.71%

scaffolds>100 	8879762	98.74%
scaffolds>500 	973482	10.82%
scaffolds>1K  	573434	6.38%
scaffolds>10K 	254225	2.83%
scaffolds>100K	14	0.00%
scaffolds>1M  	0	0.00%

Nucleotide_A	955263506	15.53%
Nucleotide_C	636996781	10.36%
Nucleotide_G	623289812	10.13%
Nucleotide_T	940316743	15.29%
GapContent_N	2994864997	48.69%
Non_ACGTN	0	0.00%
GC_Content	39.93%		(G+C)/(A+C+G+T)

N10	30211	15836
N20	21291	41073
N30	16068	72767
N40	11682	120011
N50	10654	175431
N60	10115	234866
N70	2501	337756
N80	522	943385
N90	138	3718119

NG50	113293	1
N50_scaffold-NG50_scaffold_length_difference	102639

```
Only with 180 library
```
SOAPdenovo-127mer all -s Mellotropicalis_180_only.conf -K 65 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_180_only_65mers
Import reads from file:
 /net/infofile2/2/scratch/evanslab_backups/2014_BJE3652_gDNA_seqs/Sample_Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence_09082015_untar/Sample_Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence_09082015/only_trim_paired_AND_Half/Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence_09082015_NoIndex_L006_R2_041_trim_paired.fastq.gz
Segmentation fault

SOAPdenovo-63mer all -s Mellotropicalis_180_L006_only.conf -K 61 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_180_L006_only_61mers
```
Genome size:

*[X. tropicalis](http://www.xenbase.org/anatomy/intro.do)*: 1.7 x 10^9 bp x 2 (duplicated genome) = 3.4 x 10^9

###Try to improve
If we obtain an OK assembly, we can try some "diploid" assembly program to try improving it.
- [ScaffoldScaffolder](http://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-16-S7-S5) seems promissing for dealing with diploid assembly using bubbles to identify heterozygous homologous. Should be good for sites when each subgenome is homozygous but how to deal with heteroz. sites in both/one of the sub-genomes?  
- Haplotype assembly using *X. tropicalis* as the ref genome?
 - SDhaP  
 - http://www.ncbi.nlm.nih.gov/pubmed/16077012


- Try [Platanus](http://platanus.bio.titech.ac.jp/platanus/?page_id=14) that pretends to deal with highly heterozygous? genomes
