## SOAP
```
/home/evanslab/tetra_project/Assembly/SOAP/SOAPdenovo2-src-r240/SOAPdenovo-127mer all -s /net/infofile2/2/scratch/evanslab_backups/2014_BJE3652_gDNA_seqs/Mellotropicalis.conf -K 65 -R -N 3400000000 -p 10 -V -o /net/infofile2/2/scratch/evanslab_backups/2014_BJE3652_gDNA_seqs/SOAP_Mellotropicalis_BJE3652_genome

- Error: Segmentation fault when trying to import Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence_09082015_NoIndex_L008_R2_014_trim_paired.fastq.gz

/home/evanslab/tetra_project/Assembly/SOAP/SOAPdenovo2-src-r240/SOAPdenovo-127mer all -s Mellotropicalis.conf -K 65 -R -N 3400000000 -p 10 -V -o SOAP_Mellotropicalis_BJE3652_genome 1>ass.log 2>ass.err

- Error: Ambiguous output redirect

SOAPdenovo-127mer all -s Mellotropicalis_without_180_2ndSeq.conf -K 65 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome


After: try increasing Kmer size
SOAPdenovo-127mer all -s Mellotropicalis.conf -K 80 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome

After: try decreasing Kmer size  
SOAPdenovo-63mer all -s Mellotropicalis.conf -K 63 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome
```
Results `SOAP_Mellotropicalis_BJE3652_genome` (65mers):
```
Size_includeN	5076388490
Size_withoutN	2697808415
Scaffold_Num	5507366
Mean_Size	921
Median_Size	131
Longest_Seq	148529
Shortest_Seq	100
Singleton_Num	4852830
Average_length_of_break(N)_in_scaffold	431

Known_genome_size	-794967296
Total_scaffold_length_as_percentage_of_known_genome_size	-638.57%

scaffolds>100 	5437123	98.72%
scaffolds>500 	820199	14.89%
scaffolds>1K  	494127	8.97%
scaffolds>10K 	204740	3.72%
scaffolds>100K	30	0.00%
scaffolds>1M  	0	0.00%

Nucleotide_A	820282695	16.16%
Nucleotide_C	538445828	10.61%
Nucleotide_G	530414447	10.45%
Nucleotide_T	808665445	15.93%
GapContent_N	2378580075	46.86%
Non_ACGTN	0	0.00%
GC_Content	39.62%		(G+C)/(A+C+G+T)

N10	33138	11591
N20	23425	29748
N30	19817	53560
N40	13214	85129
N50	11244	127377
N60	10358	174568
N70	6499	229794
N80	1079	467635
N90	206	1666368

NG50	148529	1
N50_scaffold-NG50_scaffold_length_difference	137285
```
Working without 180_2ndSeq but results not that good (small scaffolds): try with bigger kmer size (possible because we have a lot of data)
```
SOAPdenovo-127mer all -s Mellotropicalis_without_180_2ndSeq_125mers.conf -K 125 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_125

SOAPdenovo-127mer all -s Mellotropicalis_without_180_2ndSeq_125mers.conf -K 80 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_80
Segmentation fault
```
On iqaluk
```
/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-63mer all -s Mellotropicalis_all_libraries.conf -K 43 -m 63 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_all_libraries_43_to_63
Stuck forever in one step 

/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-127mer all -s Mellotropicalis_all_libraries.conf -K 65 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_all_libraries_65
Bus error
/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-127mer all -s Mellotropicalis_all_libraries.conf -K 67 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_all_libraries_67
/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-63mer all -s Mellotropicalis_all_libraries.conf -K 63 -R -F  -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_all_libraries_63
Segmentation fault

killed even when used the option for memory assumption -a 128

/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-63mer all -s Mellotropicalis_all_libraries_big_library_scaffoldonly.conf -K 63 -R -F -N 3500000000 -a 128 -V -p 15 -o /work/ben/Mellotropicalis_corrected_data/SOAP_assembly/2nd_try/SOAP_Mellotropicalis_BJE3652_genome_63
killed
```
Without -F
```
/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-63mer all -s Mellotropicalis_all_libraries_big_library_scaffoldonly.conf -K 63 -R -N 3500000000 -V -p 15 -o /work/ben/Mellotropicalis_corrected_data/SOAP_assembly/2nd_try/SOAP_Mellotropicalis_BJE3652_genome_63
```
Result without -F
```
Size_includeN	5907982446
Size_withoutN	3194960744
Scaffold_Num	8117825
Mean_Size	727
Median_Size	130
Longest_Seq	257805
Shortest_Seq	100
Singleton_Num	7238616
Average_length_of_break(N)_in_scaffold	334

Known_genome_size	-794967296
Total_scaffold_length_as_percentage_of_known_genome_size	-743.17%

scaffolds>100 	8020572	98.80%
scaffolds>500 	1067006	13.14%
scaffolds>1K  	564563	6.95%
scaffolds>10K 	227083	2.80%
scaffolds>100K	58	0.00%
scaffolds>1M  	0	0.00%

Nucleotide_A	966295443	16.36%
Nucleotide_C	645673442	10.93%
Nucleotide_G	632668882	10.71%
Nucleotide_T	950322977	16.09%
GapContent_N	2713021702	45.92%
Non_ACGTN	0	0.00%
GC_Content	40.01%		(G+C)/(A+C+G+T)

N10	32074	13636
N20	22060	36042
N30	17345	65371
N40	11821	108502
N50	10636	161560
N60	10072	218854
N70	1734	360259
N80	526	1022148
N90	164	3291678

NG50	257805	1
N50_scaffold-NG50_scaffold_length_difference	247169
```
K-mer 37
```
work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-63mer all -s Mellotropicalis_all_libraries_big_library_scaffoldonly.conf -K 37 -R -N 3500000000 -V -p 15 -o /work/ben/Mellotropicalis_corrected_data/SOAP_assembly/37mers_KBlib_scaff_only/SOAP_Mellotropicalis_BJE3652_genome_37
Segmentation fault
```
**Also need to try including big library (>1kb) for only scaffolding**
```
SOAPdenovo-63mer all -s Mellotropicalis_KBlib_scaff_only.conf -K 63 -R -F -N 3500000000 -V -p 15 -o /net/infofile2/2/scratch/evanslab_backups/Mellotropicalis_2016/SOAP/63mers_KBlib_scaff_only/SOAP_Mellotropicalis_BJE3652_genome_63
```
```
/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-63mer all -s Mellotropicalis_all_libraries_big_library_scaffoldonly.conf -K 63 -R -F -N 3500000000 -V -p 15 -o /work/ben/Mellotropicalis_corrected_data/SOAP_assembly/63mers_KBlib_scaff_only/SOAP_Mellotropicalis_BJE3652_genome_63
```
```
SOAPdenovo-63mer all -s Mellotropicalis_KBlib_scaff_only.conf -K 43 -m 63 -R -F -N 3500000000 -V -p 15 -o /net/infofile2/2/scratch/evanslab_backups/Mellotropicalis_2016/SOAP/63mers_KBlib_scaff_only/SOAP_Mellotropicalis_BJE3652_genome_43_63
```
**Note 09/08:** 
- Multi kmer approach use too much time & memory for big dataset. Will use the kmer size with which we have obtained the best results when we didn't consider the 2nd sequencing of 180bp library `-K 65`. And also `-K 67` (considering read length is around 100bp, `2/3 * 10 ~ 66.667` so should obtain best assembly with kmer size around that value in theory. Trying smaller value `43-63` and bigger `65-95` didn't improve the assembly (when considering the data without 2nd Sequencing og 180bp library). 
- Should use [velvet](https://github.com/dzerbino/velvet/blob/master/doc/manual_src/Manual.tex) and merge multiple assemblies obtained with different kmer size (65 & 67 at least) to compare the result. [Allpaths](http://software.broadinstitute.org/allpaths-lg/blog/?page_id=12) seems to be a good alternative option too for our dataset (but not for *Hymenochirus*):

***ALLPATHSâ€LG requires high sequence coverage of the genome in order to compensate for the shortness of the reads. The precise coverage required depends on the length and quality of the paired reads, but typically is of the order 100x or above.***

**Note 11/08:** 
- Need to try again multi-kmer approach:
 * small size better for heterozygous genome
 * big size if high proportion of repeats
-> Our genome is tetraploid (+info from jellyfish k-mer distribution): highly heteroz + repetitive

Multi k-mer approach
```
SOAPdenovo-127mer all -s Mellotropicalis_without_180_2ndSeq_125mers.conf -K 65 -m 125 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome65_to_125
(killed)

SOAPdenovo-127mer all -s Mellotropicalis_without_180_2ndSeq_125mers.conf -K 65 -m 95 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_65_to_95

SOAPdenovo-63mer all -s Mellotropicalis_without_180_2ndSeq_125mers.conf -K 43 -m 63 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_43_to_63
```
Result `SOAP_Mellotropicalis_BJE3652_genome_43_to_63`:
```
Size_includeN	6150731839
Size_withoutN	3155866842
Scaffold_Num	8993155
Mean_Size	683
Median_Size	127
Longest_Seq	113293
Shortest_Seq	100
Singleton_Num	8172338
Average_length_of_break(N)_in_scaffold	333

Known_genome_size	-794967296
Total_scaffold_length_as_percentage_of_known_genome_size	-773.71%

scaffolds>100 	8879762	98.74%
scaffolds>500 	973482	10.82%
scaffolds>1K  	573434	6.38%
scaffolds>10K 	254225	2.83%
scaffolds>100K	14	0.00%
scaffolds>1M  	0	0.00%

Nucleotide_A	955263506	15.53%
Nucleotide_C	636996781	10.36%
Nucleotide_G	623289812	10.13%
Nucleotide_T	940316743	15.29%
GapContent_N	2994864997	48.69%
Non_ACGTN	0	0.00%
GC_Content	39.93%		(G+C)/(A+C+G+T)

N10	30211	15836
N20	21291	41073
N30	16068	72767
N40	11682	120011
N50	10654	175431
N60	10115	234866
N70	2501	337756
N80	522	943385
N90	138	3718119

NG50	113293	1
N50_scaffold-NG50_scaffold_length_difference	102639

```
Only with 180 library
```
SOAPdenovo-127mer all -s Mellotropicalis_180_only.conf -K 65 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_180_only_65mers
Import reads from file:
 /net/infofile2/2/scratch/evanslab_backups/2014_BJE3652_gDNA_seqs/Sample_Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence_09082015_untar/Sample_Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence_09082015/only_trim_paired_AND_Half/Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence_09082015_NoIndex_L006_R2_041_trim_paired.fastq.gz
Segmentation fault

SOAPdenovo-63mer all -s Mellotropicalis_180_L006_only.conf -K 61 -R -N 3500000000 -V -p 10 -o SOAP_Mellotropicalis_BJE3652_genome_180_L006_only_61mers
```
Genome size:

*[X. tropicalis](http://www.xenbase.org/anatomy/intro.do)*: 1.7 x 10^9 bp x 2 (duplicated genome) = 3.4 x 10^9

### Try to run with all the data

Mark from Sharcnet increased the memory we can use because of allpaths. Let's try again
```
/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-127mer all -s Mellotropicalis_all_libraries_memory.conf -K 65 -R -F -N 3500000000 -V -p 15 -o /work/ben/Mellotropicalis_corrected_data/SOAP_assembly/SOAP_Mellotropicalis_BJE3652_genome_65_memory

/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-63mer all -s Mellotropicalis_all_libraries_memory_separate_lane_180_2.conf -K 63 -R -N 3500000000 -V -p 5 -o /work/ben/Mellotropicalis_corrected_data/SOAP_assembly/SOAP_Mellotropicalis_BJE3652_genome_63_memory

[ben@iqaluk:/work/ben/Mellotropicalis_corrected_data/SOAP_assembly] /work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-63mer all -s Mellotropicalis_all_libraries_memory_separate_lane_180_2.conf -K 33 -V -p 5 -o /work/ben/Mellotropicalis_corrected_data/SOAP_assembly/SOAP_Mellotropicalis_BJE3652_genome_33_memory

 /work/ben/Mellotropicalis_corrected_data/allpaths/data/Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence_09082015_NoIndex_L006_R2_071_trim_paired.fastq.gz
Segmentation fault

```
Recording memory usage
```
top -b -p 55018 -d 1800 >>top.out # "-b": in bash, "-p": PID number, "-d": delay in seconds 
```
```
/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-63mer all -s Mellotropicalis_all_libraries_memory_separate_lane_180_2.conf -K 33 -V -p 5 -o /work/ben/Mellotropicalis_corrected_data/SOAP_assembly/SOAP_Mellotropicalis_BJE3652_genome_33_memory >SOAP_check_memory.out
```
For some reason the last command didn't fail at loading the data... Perhaps the fact of redirecting the results (however, we weren't able at 1st tor use `1>ass.log 2>ass.err` as suggested on the SOAP webpage).
```
Done with 1138771 scaffolds, 0 gaps finished, 13446909 gaps overall.

Overall time spent on constructing scaffolds: 2590m.
Time for the whole pipeline: 7211m.
```
```
less SOAP_Mellotropicalis_BJE3652_genome_33_memory.scafStatistics

<-- Information for assembly Scaffold '/work/ben/Mellotropicalis_corrected_data/SOAP_assembly/SOAP_Mellotropicalis_BJE3652_genome_33_memory.scafSeq'.(cut_off_length < 100bp) -->

Size_includeN   5073922146
Size_withoutN   1671195048
Scaffold_Num    2654558
Mean_Size       1911
Median_Size     206
Longest_Seq     172751
Shortest_Seq    100
Singleton_Num   1515787
Average_length_of_break(N)_in_scaffold  1281

Known_genome_size       NaN
Total_scaffold_length_as_percentage_of_known_genome_size        NaN

scaffolds>100   2605266 98.14%
scaffolds>500   848053  31.95%
scaffolds>1K    541593  20.40%
scaffolds>10K   246073  9.27%
scaffolds>100K  14      0.00%
scaffolds>1M    0       0.00%

Nucleotide_A    510513933       10.06%
Nucleotide_C    328184859       6.47%
Nucleotide_G    326721233       6.44%
Nucleotide_T    505775023       9.97%
GapContent_N    3402727098      67.06%
Non_ACGTN       0       0.00%
GC_Content      39.19%          (G+C)/(A+C+G+T)

N10     32148   12151
N20     22929   30575
N30     20451   54201
N40     13674   83219
N50     11603   124323
N60     10815   169738
N70     10267   217953
N80     4952    275765
N90     908     579958

NG50    NaN     NaN
N50_scaffold-NG50_scaffold_length_difference    NaN
```
Being optimistic
```
/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-63mer all -s Mellotropicalis_all_libraries_memory.conf -K 33 -V -p 5 -R -F -o /work/ben/Mellotropicalis_corrected_data/SOAP_assembly/SOAP_Mellotropicalis_BJE3652_genome_33_memory_180together >SOAP_check_memory_all180together.out
```
Filling part crashed with `Aborted` but able to get the statistics from the console (`ctrl+A esc`) 
```
Time spent on constructing scaffolds: 11471s.

The final rank

*******************************
 Scaffold number                  1142790
 In-scaffold contig number        9881182
 Total scaffold length            4775108587
 Average scaffold length          4178
 Filled gap number                7717266
 Longest scaffold                 112002
 Scaffold and singleton number    3357171
 Scaffold and singleton length    5065714323
 Average length                   1508
 N50                              11399
 N90                              764
 Weak points                      75

*******************************
```
```
/work/ben/2016_Hymenochirus/SOAPdenovo2-src-r240/SOAPdenovo-63mer all -s Mellotropicalis_all_libraries_memory.conf -K 63 -V -p 5 -R -o /work/ben/Mellotropicalis_corrected_data/SOAP_assembly/SOAP_Mellotropicalis_BJE3652_genome_63_memory_180together >SOAP_check_memory_63_all180together.out
```
```
<-- Information for assembly Scaffold '/work/ben/Mellotropicalis_corrected_data/SOAP_assembly/SOAP_Mellotropicalis_BJE3652_genome_63_memory_180together.scafSeq'.(cut_off_length < 100bp) -->

Size_includeN   6029279388
Size_withoutN   3241624546
Scaffold_Num    8577760
Mean_Size       702
Median_Size     127
Longest_Seq     147593
Shortest_Seq    100
Singleton_Num   7666270
Average_length_of_break(N)_in_scaffold  324

Known_genome_size       NaN
Total_scaffold_length_as_percentage_of_known_genome_size        NaN

scaffolds>100   8473447 98.78%
scaffolds>500   1089910 12.71%
scaffolds>1K    578889  6.75%
scaffolds>10K   238209  2.78%
scaffolds>100K  35      0.00%
scaffolds>1M    0       0.00%

Nucleotide_A    979989873       16.25%
Nucleotide_C    656159543       10.88%
Nucleotide_G    642819722       10.66%
Nucleotide_T    962655408       15.97%
GapContent_N    2787654842      46.24%
Non_ACGTN       0       0.00%
GC_Content      40.07%          (G+C)/(A+C+G+T)

N10     30673   14857
N20     21392   39172
N30     16001   70316
N40     11573   117083
N50     10569   171874
N60     10065   230526
N70     1605    391501
N80     488     1114734
N90     154     3582345

NG50    NaN     NaN
N50_scaffold-NG50_scaffold_length_difference    NaN

```

### Multi-kmer approach - all data

(28/05)

We are more computing resources so I will try some more runs with SOAP de novo. The contig total length is fairly close to the expected genome size but not at the end of the scaffolding. For DBG2OLC we directly use the contigs so we can try to improve the contigs statistics at least. I want to try again using the multi-kmer approach using as the smallest kmer size `47` (`92/2=46`) and max `61` (`92*2/3=61`). 92-93 being the mean `Read_length_mean` estimated for [meraculous](https://github.com/caro46/Tetraploid_project/blob/master/Assembly_meraculous.md). 

We will be using the short insert reads for all the steps and the long insert libraries only for scaffolding. The max read length should be ~ 150bp as identified when using [LoRDEC](https://github.com/caro46/Tetraploid_project/blob/master/correction_pacbio.md). We want to try to use the whole length of all the libraries.  
We are using the mate pair after filtering from `nxtrim` (get rid of pe contamination in the mate libraries which could cause some trouble in the assembly process).
(submitted on graham on the 28/05).

Failed as for another during the filling part probably due to memory. Resubbmitted (31/05) the scaffolding part but following were the intermediary results from the console which are pretty encouraging (I think it is the 1st time I might obtained less than a million of scaffolds). The total length is higher than expected size probably to high polymorphism or repeats that can't be solved...Pacbio should help. 
```
The final rank

*******************************
 Scaffold number                  909826
 In-scaffold contig number        15145962
 Total scaffold length            4552662351
 Average scaffold length          5003
 Filled gap number                4983561
 Longest scaffold                 156135
 Scaffold and singleton number    8741811
 Scaffold and singleton length    5286506664
 Average length                   604
 N50                              11239
 N90                              214
 Weak points                      681
```

(08/08/18)

So after the assembly finally went through the queue it failed. Not clear if it is because a core failed again... The intermediary results were encouraging but I have never been successfully able to use the outputs from SOAPdenovo as inputs from DBG2OLC even when trying to make them "look like" meraculous outputs (i.e. changing the sequences header). So since to get better results we need to be able to take advanntage of the continuity of the pacbio reads, we need to be able to run DBG2OLC... So for now (considering also that the queue is long and that the server will be down by the end of the month), I'll leave the SOAP option for now. 

Was rerun successfully without the filling option. 

```
less Quast_SOAP_47_61_scaffolds/report.txt

All statistics are based on contigs of size >= 500 bp, unless otherwise noted (e.g., "# contigs (>= 0 bp)" and "Total length (>= 0 bp)" include all contigs).
Suggestion: all assemblies contain continuous fragments of N's of length >= 10 bp. You may consider rerunning QUAST using --scaffolds (-s) option!

Assembly                    SOAP_Mellotropicalis_BJE3652_all_libraries_big_library_scaffoldonly_47_61mers.scafSeq  SOAP_Mellotropicalis_BJE3652_genome_63.scafSeq
# contigs (>= 0 bp)         8747472                                                                                8117825
# contigs (>= 1000 bp)      511170                                                                                 565105
# contigs (>= 5000 bp)      263093                                                                                 263126
# contigs (>= 10000 bp)     222708                                                                                 227154
# contigs (>= 25000 bp)     27167                                                                                  26647
# contigs (>= 50000 bp)     2446                                                                                   2715
Total length (>= 0 bp)      5765524150                                                                             5907982446
Total length (>= 1000 bp)   4336244034                                                                             4398088021
Total length (>= 5000 bp)   3892871210                                                                             3888205495
Total length (>= 10000 bp)  3597974221                                                                             3628135653
Total length (>= 25000 bp)  970107622                                                                              963188667
Total length (>= 50000 bp)  149453914                                                                              169925946
# contigs                   889240                                                                                 1068850
Largest contig              156266                                                                                 257805
Total length                4600139242                                                                             4750326080
GC (%)                      39.29                                                                                  39.33
N50                         12246                                                                                  11783
N75                         10119                                                                                  10060
L50                         102389                                                                                 109516
L75                         208017                                                                                 220637
# N's per 100 kbp           58459.19                                                                               57023.58
```

### Gapcloser

Using `SOAP_Mellotropicalis_BJE3652_all_libraries_big_library_scaffoldonly_47_61mers.scafSeq` as input, the other input (config file) is the same as used with gapcloser. 

Submitted on 10 cpus, 500gb for 7 days. Not 1% done. 

Proposed solutions: 30cpus, 500gb for 28days (maximum) on scaffolds >= 500gb

To select only scaffolds >= 500bp:
```
awk '/^>/ {printf("\n%s\n",$0);next; } { printf("%s",$0);}  END {printf("\n");}' SOAP_Mellotropicalis_BJE3652_all_libraries_big_library_scaffoldonly_47_61mers.scafSeq >SOAP_Mellotropicalis_BJE3652_all_libraries_big_library_scaffoldonly_47_61mers_1line.scafSeq
awk '!/^>/ { next } { getline seq } length(seq) >= 500 { print $0 "\n" seq }' SOAP_Mellotropicalis_BJE3652_all_libraries_big_library_scaffoldonly_47_61mers_1line.scafSeq >SOAP_Mellotropicalis_BJE3652_all_libraries_big_library_scaffoldonly_47_61mers_scafSeq_min500bp.fa
``` 
The 2 commands were necessary because of the multi-line format of SOAPdenovo output.

Updates (12/02): 

- Gapcloser failed on the `>=500bp` subset because of memory. After 14days, 128044 of the 889240 scaffolds were done

- Submitting a job using 32 CPUS, 800Gb, 1 node, 28 days on `>=10000bp`. From the Quast results, we would be using 222708 scaffolds, representing a length of 3597974221 (still in the order of the genome size of ~3.5Gb). Good chance that this job would go through the end. After the gapclosing step, I'll run quast on the subset before and after gapclosing, we can then make a decision: using the output to continue and improve the scaffolding, try again on a larger subset (would need to have more resources for that). 


### Try to improve
If we obtain an OK assembly, we can try some "diploid" assembly program to try improving it.
- [ScaffoldScaffolder](http://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-16-S7-S5) seems promissing for dealing with diploid assembly using bubbles to identify heterozygous homologous. Should be good for sites when each subgenome is homozygous but how to deal with heteroz. sites in both/one of the sub-genomes?  
- Haplotype assembly using *X. tropicalis* as the ref genome?
 - SDhaP  
 - http://www.ncbi.nlm.nih.gov/pubmed/16077012


- Try [Platanus](http://platanus.bio.titech.ac.jp/platanus/?page_id=14) that pretends to deal with highly heterozygous? genomes


### perl sript use to create the conf file for SOAPdenovo
```perl
#!/usr/bin/perl

use strict;
use warnings;

#my $output = "/net/infofile2/2/scratch/evanslab_backups/2014_BJE3652_gDNA_seqs/Mellotropicalis";
my $output = "/work/ben/Mellotropicalis_corrected_data/SOAP_assembly/Mellotropicalis_all_libraries_memory.conf";
# Use the open() function to create the file.
unless(open FILE, '>'.$output) {
# Die with error message 
# if we can't open it.
    die "\nUnable to create $output\n";
}

##180_1stseq_lib
my @files_R1_180;
my @files_R2_180;
my $path_to_data_180;

#$path_to_data_180 = "/work/ben/Mellotropicalis_corrected_data/Sample_BenEvansBJE3652_180bp_Library_quake/";
$path_to_data_180 = "/work/ben/Mellotropicalis_corrected_data/allpaths/data/";
@files_R1_180 = glob($path_to_data_180."BenEvansBJE3652_180bp_Library_GTTTCG*_R1*_trim_paired.cor.fastq.gz");
@files_R2_180 = glob($path_to_data_180."BenEvansBJE3652_180bp_Library_GTTTCG*_R2*_trim_paired.cor.fastq.gz");


##180_2nseq_lib
my @files_R1_180_2;
my @files_R2_180_2;
my $path_to_data_180_2;

#$path_to_data_180_2 = "/work/ben/Mellotropicalis_corrected_data/Sample_Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence_09082015_trim/";
$path_to_data_180_2 = "/work/ben/Mellotropicalis_corrected_data/allpaths/data/";
@files_R1_180_2 = glob($path_to_data_180_2."Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence*_R1*_trim_paired.fastq.gz");
@files_R2_180_2 = glob($path_to_data_180_2."Ben_Evans_BJE3652_180_BP_Library_2nd_Sequence*_R2*_trim_paired.fastq.gz");

##6KB_Data_first_lane

my @files_R1_6KB;
my @files_R2_6KB;
my $path_to_data_6KB;

#$path_to_data_6KB = "/work/ben/Mellotropicalis_corrected_data/BJE3652_6KB_Data_first_lane_quake/";
$path_to_data_6KB = "/work/ben/Mellotropicalis_corrected_data/allpaths/data/";
@files_R1_6KB = glob($path_to_data_6KB."Ben-Evans-BJE3652-6kb_NoIndex*_R1*_trim_paired.cor.fastq.gz");
@files_R2_6KB = glob($path_to_data_6KB."Ben-Evans-BJE3652-6kb_NoIndex*_R2*_trim_paired.cor.fastq.gz");

##6KB_Data_2nd_Sequencing

my @files_R1_6KB_2;
my @files_R2_6KB_2;
my $path_to_data_6KB_2;

#$path_to_data_6KB_2 = "/work/ben/Mellotropicalis_corrected_data/Sample_Ben_Evans_BJE3652_6kb_2nd_Sequencing_Run_quake/";
$path_to_data_6KB_2 = "/work/ben/Mellotropicalis_corrected_data/allpaths/data/";
@files_R1_6KB_2 = glob($path_to_data_6KB_2."Ben_Evans_BJE3652_6kb_2nd_Sequencing_Run_NoIndex*_R1*_trim_paired.cor.fastq.gz");
@files_R2_6KB_2 = glob($path_to_data_6KB_2."Ben_Evans_BJE3652_6kb_2nd_Sequencing_Run_NoIndex*_R2*_trim_paired.cor.fastq.gz");

##1000bp

my @files_R1_1000bp;
my @files_R2_1000bp;
my $path_to_data_1000bp;

#$path_to_data_1000bp = "/work/ben/Mellotropicalis_corrected_data/Sample_BenEvansBJE3652_1000bp_Library_quake/";
$path_to_data_1000bp = "/work/ben/Mellotropicalis_corrected_data/allpaths/data/";
@files_R1_1000bp = glob($path_to_data_1000bp."BenEvansBJE3652_1000bp_Library*_R1*_trim_paired.cor.fastq.gz");
@files_R2_1000bp = glob($path_to_data_1000bp."BenEvansBJE3652_1000bp_Library*_R2*_trim_paired.cor.fastq.gz");

##10KB

my @files_R1_10KB;
my @files_R2_10KB;
my $path_to_data_10KB;

#$path_to_data_10KB = "/work/ben/Mellotropicalis_corrected_data/Sample_Ben_Evans_BJE3652_10kb_Mate_Pair_Library_quake/";
$path_to_data_10KB = "/work/ben/Mellotropicalis_corrected_data/allpaths/data/";
@files_R1_10KB = glob($path_to_data_10KB."Ben_Evans_BJE3652_10kb_Mate_Pair_Library_NoIndex*_R1*_trim_paired.cor.fastq.gz");
@files_R2_10KB = glob($path_to_data_10KB."Ben_Evans_BJE3652_10kb_Mate_Pair_Library_NoIndex*_R2*_trim_paired.cor.fastq.gz");

##400bp

my @files_R1_400bp;
my @files_R2_400bp;
my $path_to_data_400bp;

#$path_to_data_400bp = "/work/ben/Mellotropicalis_corrected_data/Sample_BenEvansBJE3652_400bp_Library_quake/";
$path_to_data_400bp = "/work/ben/Mellotropicalis_corrected_data/allpaths/data/";
@files_R1_400bp = glob($path_to_data_400bp."BenEvansBJE3652_400bp_Library_*_R1*_trim_paired.cor.fastq.gz");
@files_R2_400bp = glob($path_to_data_400bp."BenEvansBJE3652_400bp_Library_*_R2*_trim_paired.cor.fastq.gz");

my $y;
my $z;
my $x;
my $f;
my $g;
my $h;
my $j;

print FILE "max_rd_len=152","\n","[LIB]","\n",
        "avg_ins=180","\n", "reverse_seq=0","\n","asm_flags=3","\n",
        "rd_len_cutoff=150","\n","rank=1","\n","pair_num_cutoff=3","\n","map_len=32","\n";
for ($y=0; $y<=$#files_R1_180; $y ++) {
	print FILE #"max_rd_len=100","\n","[LIB]","\n",
	#"avg_ins=180","\n", "reverse_seq=0","\n","asm_flags=3","\n",
	#"rd_len_cutoff=100","\n","rank=1","\n","pair_num_cutoff=3","\n","map_len=32","\n",
	"q1=",$files_R1_180[$y],"\n","q2=",$files_R2_180[$y],"\n";
}
print FILE "[LIB]","\n","avg_ins=180","\n", "reverse_seq=0","\n","asm_flags=3","\n",
        "rd_len_cutoff=150","\n","rank=2","\n","pair_num_cutoff=3","\n","map_len=32","\n";
for ($z=0; $z<=$#files_R1_180_2; $z ++) {
	print FILE #"[LIB]","\n","avg_ins=180","\n", "reverse_seq=0","\n","asm_flags=3","\n",
	#"rd_len_cutoff=100","\n","rank=1","\n","pair_num_cutoff=3","\n","map_len=32","\n",
        "q1=",$files_R1_180_2[$z],"\n","q2=",$files_R2_180_2[$z],"\n";
}
print FILE "[LIB]","\n","avg_ins=400","\n", "reverse_seq=0","\n","asm_flags=3","\n",
        "rd_len_cutoff=150","\n","rank=3","\n","pair_num_cutoff=3","\n","map_len=32","\n";
for ($j=0; $j<=$#files_R1_400bp; $j ++){
	print FILE #"[LIB]","\n","avg_ins=400","\n", "reverse_seq=0","\n","asm_flags=3","\n",
	#"rd_len_cutoff=100","\n","rank=2","\n","pair_num_cutoff=3","\n","map_len=32","\n",
	"q1=",$files_R1_400bp[$j],"\n","q2=",$files_R2_400bp[$j],"\n";
}
print FILE "[LIB]","\n","avg_ins=1000","\n", "reverse_seq=0","\n","asm_flags=3","\n",
        "rd_len_cutoff=150","\n","rank=4","\n","pair_num_cutoff=5","\n","map_len=35","\n";
for ($g=0; $g<=$#files_R1_1000bp; $g ++){
	print FILE #"[LIB]","\n","avg_ins=1000","\n", "reverse_seq=0","\n","asm_flags=3","\n",
	#"rd_len_cutoff=100","\n","rank=3","\n","pair_num_cutoff=3","\n","map_len=32","\n",
	"q1=",$files_R1_1000bp[$g],"\n","q2=",$files_R2_1000bp[$g],"\n";
}
print FILE "[LIB]","\n","avg_ins=6000","\n", "reverse_seq=1","\n","asm_flags=3","\n",
        "rd_len_cutoff=150","\n","rank=5","\n","pair_num_cutoff=5","\n","map_len=35","\n";
for ($x=0; $x<=$#files_R1_6KB; $x ++){
	print FILE #"[LIB]","\n","avg_ins=6000","\n", "reverse_seq=0","\n","asm_flags=3","\n",
	#"rd_len_cutoff=100","\n","rank=4","\n","pair_num_cutoff=3","\n","map_len=32","\n",
	"q1=",$files_R1_6KB[$x],"\n","q2=",$files_R2_6KB[$x],"\n";
}
print FILE "[LIB]","\n","avg_ins=6000","\n", "reverse_seq=1","\n","asm_flags=3","\n",
        "rd_len_cutoff=150","\n","rank=5","\n","pair_num_cutoff=5","\n","map_len=35","\n";
for ($f=0; $f<=$#files_R1_6KB_2; $f ++){
	print FILE #"[LIB]","\n","avg_ins=6000","\n", "reverse_seq=0","\n","asm_flags=3","\n",
	#"rd_len_cutoff=100","\n","rank=4","\n","pair_num_cutoff=3","\n","map_len=32","\n",
	"q1=",$files_R1_6KB_2[$f],"\n","q2=",$files_R2_6KB_2[$f],"\n";
}
print FILE "[LIB]","\n","avg_ins=10000","\n", "reverse_seq=1","\n","asm_flags=3","\n",
        "rd_len_cutoff=150","\n","rank=6","\n","pair_num_cutoff=5","\n","map_len=35","\n";
for ($h=0; $h<=$#files_R1_10KB; $h ++){
	print FILE #"[LIB]","\n","avg_ins=10000","\n", "reverse_seq=0","\n","asm_flags=3","\n",
	#"rd_len_cutoff=100","\n","rank=5","\n","pair_num_cutoff=3","\n","map_len=32","\n",
	"q1=",$files_R1_10KB[$h],"\n","q2=",$files_R2_10KB[$h],"\n";
}

# close the file.
close FILE;
    exit 0;

```
