# Download Quickmerge
```
git clone https://github.com/mahulchak/quickmerge
cd quickmerge/
bash make_merger.sh
export PATH=/work/cauretc/programs/quickmerge:/work/cauretc/programs/quickmerge/MUMmer3.23:$PATH
```
It was not obvious for me what is considered as reference and query. Apparently the `hybrid_assembly_fasta` is the reference and `self_assembly_fasta` is the query. So the query needs to be specified 1st, then the reference (I 1st did the opposite - updated the good commands on Jan.17). 

```
usage: merge_wrapper.py [-h] [-pre PREFIX] [-hco HCO] [-c C]
                        [-l LENGTH_CUTOFF] [--no_nucmer] [--no_delta]
                        [--stop_after_nucmer] [--stop_after_df]
                        [-lm LENGTH_MINIMUM] [--clean_only]
                        hybrid_assembly_fasta self_assembly_fasta
```
- `-hco`: controls the overlap cutoff used in selection of anchor contigs. Default is 5.0.

- `-l`: controls the length cutoff for anchor contigs. A good rule of thumb is to start with the N50 of the self assembly. E.g. if the N50 of your self assembly is 2Mb then use 2000000 as your cutoff. Lowering this value may lead to more merging but may increase the probability of mis-joins.

- `-lm`: controls the minimum alignment length to be considered for merging. This is especially helpful for repeat-rich genomes. Default is 0 but higher values (>5000) are recommended.

For our Allpaths assembly: `N50~1920`. So we should for future runs use `-l 1900` and `-lm 1000` (since our `N50` is pretty small). The 1st runs in paragraphs `Allpaths - SOAP` and `Allpaths - backbone` have been run with default to see the memory usage, time and checking on the correct running of the program.

# Allpaths - SOAP

Allpaths has "small" but good scaffolds, SOAP has longer sequences but not great (bunch of missing data) - let see if there is still some improvement by using these 2 and how the program behaves
```
## need to unzip 1st
gunzip /work/cauretc/2017_Mellotropicalis/pseudomolecules/allpaths/final.assembly.fasta.gz
gunzip /work/cauretc/2017_Mellotropicalis/SOAP_assembly/SOAP_Mellotropicalis_BJE3652_genome_33_memory.scafSeq.gz
## running the command with default parameters
merge_wrapper.py /work/cauretc/2017_Mellotropicalis/pseudomolecules/allpaths/final.assembly.fasta /work/cauretc/2017_Mellotropicalis/SOAP_assembly/SOAP_Mellotropicalis_BJE3652_genome_33_memory.scafSeq
## some info about time
# CONSTRUCTIONTIME /work/cauretc/programs/quickmerge/MUMmer3.23/mummer out.ntref 3528.04
# COMPLETETIME /work/cauretc/programs/quickmerge/MUMmer3.23/mummer out.ntref 5529.16
# SPACE /work/cauretc/programs/quickmerge/MUMmer3.23/mummer out.ntref 5506.50
```
Hum... easy to run... Kinda want to give a shot using the backbone as ref... even f I might have some chimeras stuff (couldn't run the last step of DBG2OLC because of the HUGE number of files being produced).

# Allpaths - backbone

Need to keep in mind that probably issue with the DBG2OLC backbone file. But the output should be more like the allpaths one, the backbone is kinda just to use it to order the allpaths scaffolds. So in theory should be good (if conflict between both assemblies, Allpaths should win).
```
merge_wrapper.py /work/cauretc/2017_Mellotropicalis/pseudomolecules/allpaths/final.assembly.fasta /work/cauretc/2017_Mellotropicalis/pseudomolecules/backbone_raw.fasta
## some info about time
# CONSTRUCTIONTIME /work/cauretc/programs/quickmerge/MUMmer3.23/mummer out.ntref 4159.21
# COMPLETETIME /work/cauretc/programs/quickmerge/MUMmer3.23/mummer out.ntref 6278.10
# SPACE /work/cauretc/programs/quickmerge/MUMmer3.23/mummer out.ntref 3491.72
```
When I'll do a real one - if possible using canu - should probably first try to run something to close the gaps.

## Quast - 1st run
```
/work/ben/quast-4.5/quast.py backbone_allpaths/merged.fasta /work/cauretc/2017_Mellotropicalis/pseudomolecules/allpaths/final.assembly.fasta -o /work/cauretc/2017_Mellotropicalis/quickmerge_analysis/backbone_allpaths/quast_results/

less report.txt

All statistics are based on contigs of size >= 500 bp, unless otherwise noted (e.g., "# contigs (>= 0 bp)" and "Total length (>= 0 bp)" include all contigs).
Suggestion: all assemblies contain continuous fragments of N's of length >= 10 bp. You may consider rerunning QUAST using --scaffolds (-s) option!

Assembly                    merged     final.assembly
# contigs (>= 0 bp)         320104     320227        
# contigs (>= 1000 bp)      286125     286034        
# contigs (>= 5000 bp)      27059      19292         
# contigs (>= 10000 bp)     8600       1887          
# contigs (>= 25000 bp)     1274       6             
# contigs (>= 50000 bp)     8          0             
Total length (>= 0 bp)      732585008  607138367     
Total length (>= 1000 bp)   700024110  574372221     
Total length (>= 5000 bp)   275464847  136492195     
Total length (>= 10000 bp)  153811082  23462309      
Total length (>= 25000 bp)  39144451   159305        
Total length (>= 50000 bp)  432268     0             
# contigs                   320104     320227        
Largest contig              61708      30521         
Total length                732585008  607138367     
GC (%)                      39.38      38.79         
N50                         2821       1920          
N75                         1339       1243          
L50                         51722      76909         
L75                         154426     178611        
# N's per 100 kbp           5404.83    6522.06
```
