# Download Quickmerge
```
git clone https://github.com/mahulchak/quickmerge
cd quickmerge/
bash make_merger.sh
export PATH=/work/cauretc/programs/quickmerge:/work/cauretc/programs/quickmerge/MUMmer3.23:$PATH
```
It was not obvious for me what is considered as reference and query. Apparently the `hybrid_assembly_fasta` is the query and `self_assembly_fasta` is the reference. So the query needs to be specified 1st, then the reference (I 1st did the opposite - updated the good commands on Jan.17). 

```
usage: merge_wrapper.py [-h] [-pre PREFIX] [-hco HCO] [-c C]
                        [-l LENGTH_CUTOFF] [--no_nucmer] [--no_delta]
                        [--stop_after_nucmer] [--stop_after_df]
                        [-lm LENGTH_MINIMUM] [--clean_only]
                        hybrid_assembly_fasta self_assembly_fasta
```
- `-hco`: controls the overlap cutoff used in selection of anchor contigs. Default is 5.0.

- `-l`: controls the length cutoff for anchor contigs. A good rule of thumb is to start with the N50 of the self assembly. E.g. if the N50 of your self assembly is 2Mb then use 2000000 as your cutoff. Lowering this value may lead to more merging but may increase the probability of mis-joins.

- `-lm`: controls the minimum alignment length to be considered for merging. This is especially helpful for repeat-rich genomes. Default is 0 but higher values (>5000) are recommended.

For our Allpaths assembly: `N50~1920`. So we should for future runs use `-l 1900` and `-lm 1000` (since our `N50` is pretty small). The 1st runs in paragraphs `Allpaths - SOAP` and `Allpaths - backbone` have been run with default to see the memory usage, time and checking on the correct running of the program.

# Allpaths - SOAP

Allpaths has "small" but good scaffolds, SOAP has longer sequences but not great (bunch of missing data) - let see if there is still some improvement by using these 2 and how the program behaves
```
## need to unzip 1st
gunzip /work/cauretc/2017_Mellotropicalis/pseudomolecules/allpaths/final.assembly.fasta.gz
gunzip /work/cauretc/2017_Mellotropicalis/SOAP_assembly/SOAP_Mellotropicalis_BJE3652_genome_33_memory.scafSeq.gz
## running the command with default parameters
merge_wrapper.py /work/cauretc/2017_Mellotropicalis/pseudomolecules/allpaths/final.assembly.fasta /work/cauretc/2017_Mellotropicalis/SOAP_assembly/SOAP_Mellotropicalis_BJE3652_genome_33_memory.scafSeq
## some info about time
# CONSTRUCTIONTIME /work/cauretc/programs/quickmerge/MUMmer3.23/mummer out.ntref 3528.04
# COMPLETETIME /work/cauretc/programs/quickmerge/MUMmer3.23/mummer out.ntref 5529.16
# SPACE /work/cauretc/programs/quickmerge/MUMmer3.23/mummer out.ntref 5506.50
```
Hum... easy to run... Kinda want to give a shot using the backbone as ref... even f I might have some chimeras stuff (couldn't run the last step of DBG2OLC because of the HUGE number of files being produced).

## Quast - 1st run

```
 /work/ben/quast-4.5/quast.py /work/cauretc/2017_Mellotropicalis/quickmerge_analysis/SOAP_allpaths/merged.fasta /work/cauretc/2017_Mellotropicalis/pseudomolecules/allpaths/final.assembly.fasta /work/cauretc/2017_Mellotropicalis/SOAP_assembly/SOAP_Mellotropicalis_BJE3652_genome_33_memory.scafSeq -o /work/cauretc/2017_Mellotropicalis/quickmerge_analysis/SOAP_allpaths/quast_results

less report.txt

All statistics are based on contigs of size >= 500 bp, unless otherwise noted (e.g., "# contigs (>= 0 bp)" and "Total length (>= 0 bp)" include all contigs).
Suggestion: all assemblies contain continuous fragments of N's of length >= 10 bp. You may consider rerunning QUAST using --scaffolds (-s) option!

Assembly                    merged     final.assembly  SOAP_Mellotropicalis_BJE3652_genome_33_memory.scafSeq
# contigs (>= 0 bp)         318659     320227          2654558                                              
# contigs (>= 1000 bp)      286647     286034          541958                                               
# contigs (>= 5000 bp)      28604      19292           275558                                               
# contigs (>= 10000 bp)     10657      1887            246283                                               
# contigs (>= 25000 bp)     2154       6               25021                                                
# contigs (>= 50000 bp)     278        0               1980                                                 
Total length (>= 0 bp)      783156318  607138367       5073922146                                           
Total length (>= 1000 bp)   752475646  574372221       4530331488                                           
Total length (>= 5000 bp)   323656931  136492195       4058112063                                           
Total length (>= 10000 bp)  206283206  23462309        3838894191                                           
Total length (>= 25000 bp)  81261596   159305          882707243                                            
Total length (>= 50000 bp)  17348626   0               117897095                                            
# contigs                   318659     320227          848955                                               
Largest contig              172751     30521           172751                                               
Total length                783156318  607138367       4749065556                                           
GC (%)                      38.78      38.79           38.96                                                
N50                         3271       1920            11981                                                
N75                         1406       1243            10258                                                
L50                         45501      76909           110537                                               
L75                         145183     178611          218933                                               
# N's per 100 kbp           21735.33   6522.06         71236.08
```

# Allpaths - backbone

Need to keep in mind that probably issue with the DBG2OLC backbone file. But the output should be more like the allpaths one, the backbone is kinda just to use it to order the allpaths scaffolds. So in theory should be good (if conflict between both assemblies, Allpaths should win).
```
merge_wrapper.py /work/cauretc/2017_Mellotropicalis/pseudomolecules/allpaths/final.assembly.fasta /work/cauretc/2017_Mellotropicalis/pseudomolecules/backbone_raw.fasta
## some info about time
# CONSTRUCTIONTIME /work/cauretc/programs/quickmerge/MUMmer3.23/mummer out.ntref 4159.21
# COMPLETETIME /work/cauretc/programs/quickmerge/MUMmer3.23/mummer out.ntref 6278.10
# SPACE /work/cauretc/programs/quickmerge/MUMmer3.23/mummer out.ntref 3491.72
```
When I'll do a real one - if possible using canu - should probably first try to run something to close the gaps.

## Quast - 1st run
```
/work/ben/quast-4.5/quast.py backbone_allpaths/merged.fasta /work/cauretc/2017_Mellotropicalis/pseudomolecules/allpaths/final.assembly.fasta -o /work/cauretc/2017_Mellotropicalis/quickmerge_analysis/backbone_allpaths/quast_results/

less report.txt

All statistics are based on contigs of size >= 500 bp, unless otherwise noted (e.g., "# contigs (>= 0 bp)" and "Total length (>= 0 bp)" include all contigs).
Suggestion: assemblies merged, final.assembly contain continuous fragments of N's of length >= 10 bp. You may consider rerunning QUAST using --scaffolds (-s) option!

Assembly                    merged     final.assembly  backbone_raw
# contigs (>= 0 bp)         320104     320227          197362      
# contigs (>= 1000 bp)      286125     286034          197309      
# contigs (>= 5000 bp)      27059      19292           191606      
# contigs (>= 10000 bp)     8600       1887            150797      
# contigs (>= 25000 bp)     1274       6               17016       
# contigs (>= 50000 bp)     8          0               80          
Total length (>= 0 bp)      732585008  607138367       2998370399  
Total length (>= 1000 bp)   700024110  574372221       2998337103  
Total length (>= 5000 bp)   275464847  136492195       2976186620  
Total length (>= 10000 bp)  153811082  23462309        2654677125  
Total length (>= 25000 bp)  39144451   159305          507668773   
Total length (>= 50000 bp)  432268     0               4393666     
# contigs                   320104     320227          197345      
Largest contig              61708      30521           68950       
Total length                732585008  607138367       2998364467  
GC (%)                      39.38      38.79           42.63       
N50                         2821       1920            17259       
N75                         1339       1243            12914       
L50                         51722      76909           65494       
L75                         154426     178611          115454      
# N's per 100 kbp           5404.83    6522.06         0.03 
```

# Allpaths - miniasm

Run on 13/02/18.

## Run 1 - default parameters
```
export PATH=/work/cauretc/programs/quickmerge/:$PATH
export PATH=/work/cauretc/programs/quickmerge/MUMmer3.23:$PATH

/work/cauretc/programs/quickmerge/merge_wrapper.py /work/cauretc/2017_Mellotropicalis/pseudomolecules/allpaths/final.assembly.fasta /work/cauretc/2017_Mellotropicalis/quickmerge_analysis/Only_pacbio_mellotrop.racon2.fa -pre allpaths_miniasm
```
### Quast
```
/work/ben/quast-4.5/quast.py /work/cauretc/2017_Mellotropicalis/quickmerge_analysis/allpaths_miniasm/merged.fasta /work/cauretc/2017_Mellotropicalis/quickmerge_analysis/Only_pacbio_mellotrop.racon2.fa /work/cauretc/2017_Mellotropicalis/pseudomolecules/allpaths/final.assembly.fasta -o /work/cauretc/2017_Mellotropicalis/quickmerge_analysis/allpaths_miniasm/quast_results/


All statistics are based on contigs of size >= 500 bp, unless otherwise noted (e.g., "# contigs (>= 0 bp)" and "Total length (>= 0 bp)" include all contigs).
Suggestion: assemblies merged, final.assembly contain continuous fragments of N's of length >= 10 bp. You may consider rerunning QUAST using --scaffolds (-s) option!

Assembly                    merged     Only_pacbio_mellotrop.racon2  final.assembly
# contigs (>= 0 bp)         320223     4483                          320227        
# contigs (>= 1000 bp)      286032     3422                          286034        
# contigs (>= 5000 bp)      19302      1262                          19292         
# contigs (>= 10000 bp)     1891       628                           1887          
# contigs (>= 25000 bp)     7          25                            6             
# contigs (>= 50000 bp)     0          2                             0             
Total length (>= 0 bp)      607274268  20549620                      607138367     
Total length (>= 1000 bp)   574510084  19826155                      574372221     
Total length (>= 5000 bp)   136620834  14631285                      136492195     
Total length (>= 10000 bp)  23548411   10081460                      23462309      
Total length (>= 25000 bp)  184710     839429                        159305        
Total length (>= 50000 bp)  0          112406                        0             
# contigs                   320223     4258                          320227        
Largest contig              30521      59729                         30521         
Total length                607274268  20460393                      607138367     
GC (%)                      38.79      40.68                         38.79         
N50                         1921       9814                          1920          
N75                         1243       4363                          1243          
L50                         76878      644                           76909         
L75                         178582     1417                          178611        
# N's per 100 kbp           6520.60    0.00                          6522.06 
```
No improvement.

## Run 2
```
/work/cauretc/programs/quickmerge/merge_wrapper.py /work/cauretc/2017_Mellotropicalis/pseudomolecules/allpaths/final.assembly.fasta /work/cauretc/2017_Mellotropicalis/quickmerge_analysis/Only_pacbio_mellotrop.racon2.fa -l 9814 -lm 5000
```
### Quast
```
/work/ben/quast-4.5/quast.py /work/cauretc/2017_Mellotropicalis/quickmerge_analysis/allpaths_miniasm/merge2/merged.fasta /work/cauretc/2017_Mellotropicalis/quickmerge_analysis/Only_pacbio_mellotrop.racon2.fa /work/cauretc/2017_Mellotropicalis/pseudomolecules/allpaths/final.assembly.fasta -o /work/cauretc/2017_Mellotropicalis/quickmerge_analysis/allpaths_miniasm/merge2/quast_results/
```
Same.

Note: need to 1st close the gaps from allpaths assembly (submitted gapcloser on cedar on 13/02).
