# Download Quickmerge
```
git clone https://github.com/mahulchak/quickmerge
cd quickmerge/
bash make_merger.sh
export PATH=/work/cauretc/programs/quickmerge:/work/cauretc/programs/quickmerge/MUMmer3.23:$PATH
```
It was not obvious for me what is considered as reference and query. Apparently the `hybrid_assembly_fasta` is the reference and `self_assembly_fasta` is the query. So the query needs to be specified 1st, then the reference (I 1st did the opposite - updated the good commands on Jan.17). 

```
usage: merge_wrapper.py [-h] [-pre PREFIX] [-hco HCO] [-c C]
                        [-l LENGTH_CUTOFF] [--no_nucmer] [--no_delta]
                        [--stop_after_nucmer] [--stop_after_df]
                        [-lm LENGTH_MINIMUM] [--clean_only]
                        hybrid_assembly_fasta self_assembly_fasta
```
- `-hco`: controls the overlap cutoff used in selection of anchor contigs. Default is 5.0.

- `-l`: controls the length cutoff for anchor contigs. A good rule of thumb is to start with the N50 of the self assembly. E.g. if the N50 of your self assembly is 2Mb then use 2000000 as your cutoff. Lowering this value may lead to more merging but may increase the probability of mis-joins.

- `-lm`: controls the minimum alignment length to be considered for merging. This is especially helpful for repeat-rich genomes. Default is 0 but higher values (>5000) are recommended.


# Allpaths - SOAP

Allpaths has "small" but good scaffolds, SOAP has longer sequences but not great (bunch of missing data) - let see if there is still some improvement by using these 2 and how the program behaves
```
## need to unzip 1st
gunzip /work/cauretc/2017_Mellotropicalis/pseudomolecules/allpaths/final.assembly.fasta.gz
gunzip /work/cauretc/2017_Mellotropicalis/SOAP_assembly/SOAP_Mellotropicalis_BJE3652_genome_33_memory.scafSeq.gz
## running the command with default parameters
merge_wrapper.py /work/cauretc/2017_Mellotropicalis/pseudomolecules/allpaths/final.assembly.fasta /work/cauretc/2017_Mellotropicalis/SOAP_assembly/SOAP_Mellotropicalis_BJE3652_genome_33_memory.scafSeq
```
Hum... easy to run... Kinda want to give a shot using the backbone as ref... even f I might have some chimeras stuff (couldn't run the last step of DBG2OLC because of the HUGE number of files being produced).

# Allpaths - backbone

Need to keep in mind that probably issue with the DBG2OLC backbone file. But the output should be more like the allpaths one, the backbone is kinda just to use it to order the allpaths scaffolds. So in theory should be good (if conflict between both assemblies, Allpaths should win).
```
merge_wrapper.py /work/cauretc/2017_Mellotropicalis/pseudomolecules/allpaths/final.assembly.fasta /work/cauretc/2017_Mellotropicalis/pseudomolecules/backbone_raw.fasta
```
When I'll do a real one - if possible using canu - should probably first try to run something to close the gaps.
