# SOAPdenovo2 GapCloser

```
Version:
	1.12

Contact:
	soap@genomics.org.cn

Usage:
	GapCloser [options]
	-a	<string>	input scaffold file name, required.
	-b	<string>	input library info file name, required.
	-o	<string>	output file name, required.
	-l	<int>		maximum read length (<=155), default=100.
	-p	<int>		overlap param(<=31), default=25.
	-t	<int>		thread number, default=1.
	-h	-?		output help information.
```

```
/home/cauretc/project/cauretc/programs/SOAPdenovo2-src-r240/GapCloser -b /home/cauretc/scratch/gap_closer/input_SOAP_conf_no180_2ndlib -a /home/cauretc/scratch/gap_closer/merged.fasta.gz -o /home/cauretc/scratch/gap_closer/merged_SOAP33memory_allpaths_gapclosed
```
The `/home/cauretc/scratch/gap_closer/merged.fasta.gz` was copied from iqaluk and made previously using `Allpaths` and `SOAP_33_memory` assemblies and `quickmerge`. The `input_SOAP_conf_no180_2ndlib` was made using the script [make_SOAP_input.pl](https://github.com/caro46/Tetraploid_project/tree/master/some_scripts/make_SOAP_input.pl).

Went out of memory (22/01/18) but except that everything seemed to work fine. I'll allow for more memory and will use only small/medium size libraries (i.e 180 - 1st library, 400 and 1000bp) - will try on running for 1 day with 200Gb memory (some nodes have a lot of memory) - directly started on cedar. OK so finished without error in ~6h but the fasta headers are weird. Not sure what is the issue, I'll try with unzipping the fasta assembly.

Checked the output file during the run and the headers seem fine now (25/01/18) so the assembly needs to be unzipped but it is ok to have gzipped fastq reads.

# Alternatives

Best alternative for now (23/01/18): GapFiller. Let's wait to see GapCloser outputs.

### [GMcloser](https://academic.oup.com/bioinformatics/article/31/23/3733/209212) 
Writers say it is more accurate however uses more memory and longer time than `GapCloser`... More memory seem a lot... So better seems to try to run correctly `GapCloser` then `bwa` and consensus.

### [Sealer](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4515008/)
To install it, it looks like we need to install abyss again (if I want to use on cedar)... And not sure if we can actually use multiple reads inputs (see command below) or if it is possible not sure if the program is made to be efficient with a lot of inputs (couldn't get abyss to finish for a mellotrop assembly and Jared was working on running abyss2) so for now I don't think it is a good alternative.

```
abyss-sealer -b <Bloom filter size> -k <kmer size> -k <kmer size>... -o <output_prefix> -S <path to scaffold file> [options]... <reads1> [reads2]...
```

### [PBjelly](https://sourceforge.net/projects/pb-jelly/)
Uses Pacbio reads to close the gaps. Problem being when I tried `remove chimeras` for DBG2OLC, nothing really was left. The software seems to also produce a lot of errors and also requires "high-confidence draft assembles" (see to download and for the reference [here](https://sourceforge.net/projects/pb-jelly/)). Not sure what they mean but I don't consider our draft assembly as "high-confidence".

### [IMAGE](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2884544/)
Tested on pretty small genomes

### [GapFiller](https://genomebiology.biomedcentral.com/articles/10.1186/gb-2012-13-6-r56)
Tested on pretty small genomes - can be downloaded [here](https://sourceforge.net/projects/gapfiller/). Was used for tree assemblies (see [Tang et al. 2016](http://agritrop.cirad.fr/580933/1/Tang%202016%20Rubber%20Tree%20Genome.pdf)) and fish ([Valenzano et al. 2015](https://www.sciencedirect.com/science/article/pii/S0092867415014889)).
