# Download and references

See [Li 2016](https://academic.oup.com/bioinformatics/article/32/14/2103/1742895#95424972) for `miniasm` and `minimap`'s publication and [Vaser et al. 2017](http://genome.cshlp.org/content/27/5/737.full) to justify pipeline `minimap`/`miniasm`/`racon`. Li 2016 and Vaser et al. 2017 highlight that miniasm is not currently adapted for long repeat genome and can cause crashes due to high memory use... However not sure how it can be worse than other softwares... Some people are trying to use it for larger genome, like plants, at least for parts of a pipeline, in this [issue](https://github.com/lh3/miniasm/issues/3) 

*It took minimap about 18 hours to find all the overlaps between 5,852,650 reads (52.6 Gbases). Thats really quick!*

See the results on a [fish assembly](http://genoweb.toulouse.inra.fr/~klopp/Ljubljana2017/Pacbio_assembly_Ljubljana2.pdf), a lab also tried for the [oyster assembly](https://faculty.washington.edu/sr320/?p=13602). Still smaller genome than ours but might be some hope... Vaser et al. 2017 hilighted the fact that `miniasm` can be useful for large genomes...So we should at least try.

[Li's github page](https://github.com/lh3/miniasm)

Downloaded like specify on the github page:
```
# Install minimap and miniasm (requiring gcc and zlib)
git clone https://github.com/lh3/minimap && (cd minimap && make) #0.2-r124-dirty
git clone https://github.com/lh3/miniasm && (cd miniasm && make) #0.2-r168-dirty
```

So there is a younger version `minimap2` that I would try since it can directly handle gzipped files and has probably some bugs fixed.
The [manual](https://lh3.github.io/minimap2/minimap2.html) for specific options, [github page](https://github.com/lh3/minimap2) from where it has been downloaded

```
git clone https://github.com/lh3/minimap2
cd minimap2 && make
```

And `racon` ([see](http://inf-biox121.readthedocs.io/en/2017/Assembly/practicals/07_Assembly_using_minasm+racon.html) for summary of commands).

See `racon`'s [github page](https://github.com/isovic/racon)
```
git clone https://github.com/isovic/racon.git && cd racon && make modules && make tools && make -j
```

The issue with the new `.bam` Sequel format is that there is not possibility of obtaining a per-base score necessary for a `.fastq` format. Racon requires reads to be in a `fastq` format. It was treated as an issue [here on the github page](https://github.com/isovic/racon/issues/28) - summary of the discussion:

*Hi,
Does racon support fasta as reads rather than fastq files?
Thank you in advance.*

*Hello Michal,
Racon was designed to use base quality values for better accuracy. If you do not have them, transform your FASTA file into FASTQ with uniform dummy qualities (for example, set them to '\*' and disable the quality filter with '--bq -1').
Best regards,
Robert*

*Hi
I would like to pitch in here. I am struggling for a large genome with memory issues as all my reads (400GB) are too large for the memory. Obviously, if I remove the quality information this shrinks to <200GB.
Regarding the fact that all newer PacBio (Sequel) data do not contain quality values anymore FASTA support would be really great. By providing such an option directly in Racon instead of generating then dummy quality values instead one would save a lot of time, space and memory.
What do you think?*

*Hello,
sorry for the delayed response. Racon was designed to use base quality values and leaving them out might yield lower accuracy. We are in the middle of a refactor and we are considering to enable FASTA reads as input, but more evaluation is needed.
Best regards,
Robert*

# Commands

## Example
```
# Overlap
minimap/minimap -Sw5 -L100 -m0 -t8 reads.fq reads.fq | gzip -1 > reads.paf.gz
# Layout
miniasm/miniasm -f reads.fq reads.paf.gz > reads.gfa
```
```
racon -t num_threads reads.fastq mapped_reads.paf assembly.fasta consensus.fasta
```
## Format

Reads can be `.fastq` or `.fasta` ([see](https://github.com/lh3/miniasm/blob/master/miniasm.1)).

The output `.gfa` can be converted into a `.fasta` using something similar as this [page](http://inf-biox121.readthedocs.io/en/2017/Assembly/practicals/07_Assembly_using_minasm+racon.html):
```
gunzip -c assembly.gfa.gz | awk '{print ">"$2; print $3}' | gzip > raw_assembly.fasta
```
## Usage

### minimap
```
../minimap/minimap 
Usage: minimap [options] <target.fa> [query.fa] [...]
Options:
  Indexing:
    -k INT     k-mer size [15]
    -w INT     minizer window size [{-k}*2/3]
    -I NUM     split index for every ~NUM input bases [4G]
    -d FILE    dump index to FILE []
    -l         the 1st argument is a index file (overriding -k, -w and -I)
  Mapping:
    -f FLOAT   filter out top FLOAT fraction of repetitive minimizers [0.001]
    -r INT     bandwidth [500]
    -m FLOAT   merge two chains if FLOAT fraction of minimizers are shared [0.50]
    -c INT     retain a mapping if it consists of >=INT minimizers [4]
    -L INT     min matching length [40]
    -g INT     split a mapping if there is a gap longer than INT [10000]
    -T INT     SDUST threshold; 0 to disable SDUST [0]
    -S         skip self and dual mappings
    -O         drop isolated hits before chaining (EXPERIMENTAL)
    -P         filtering potential repeats after mapping (EXPERIMENTAL)
    -x STR     preset (recommended to be applied before other options) []
               ava10k: -Sw5 -L100 -m0 (PacBio/ONT all-vs-all read mapping)
  Input/Output:
    -t INT     number of threads [3]
    -V         show version number

```
### miniasm
```
Usage: miniasm [options] <in.paf>
Options:
  Pre-selection:
    -R          prefilter clearly contained reads (2-pass required)
    -m INT      min match length [100]
    -i FLOAT    min identity [0.05]
    -s INT      min span [2000]
    -c INT      min coverage [3]
  Overlap:
    -o INT      min overlap [same as -s]
    -h INT      max over hang length [0]
    -I FLOAT    min end-to-end match ratio [0.8]
  Layout:
    -g INT      max gap differences between reads for trans-reduction [1000]
    -d INT      max distance for bubble popping [50000]
    -e INT      small unitig threshold [4]
    -f FILE     read sequences []
    -n INT      rounds of short overlap removal [3]
    -r FLOAT[,FLOAT]
                max and min overlap drop ratio [0.7,0.5]
    -F FLOAT    aggressive overlap drop ratio in the end [0.8]
  Miscellaneous:
    -p STR      output information: bed, paf, sg or ug [ug]
    -b          both directions of an arc are present in input
    -1          skip 1-pass read selection
    -2          skip 2-pass read selection
    -V          print version number
```
### racon
#### Convert pacbio to fastq
Using `reformat.sh` from `BBMap_37.36`. Since the script requires other modules of `bbmap`, I copied the `BJE3652.all.subreads.fasta.gz` over to my `iqaluk` account and run the command there, in my `scratch`, then I'll move the output to my `archive` and on `cedar`.
```
reformat.sh in=pbio_input.bam out=stdout.fa | reformat.sh in=stdin.fa out=new.fq.gz qfake=30
/work/cauretc/programs/bbmap/reformat.sh in=BJE3652.all.subreads.fasta.gz out=BJE3652.all.subreads30qfake.fq.gz qfake=30
```
```
Input is being processed as unpaired
Input:                          4982901 reads           40644849469 bases
Output:                         4982901 reads (100.00%)         40644849469 bases (100.00%)

Time:                           309.922 seconds.
Reads Processed:       4982k    16.08k reads/sec
Bases Processed:      40644m    131.15m bases/sec
```
When checking the output, didn't seem to work. Let's try directly from `.bam`
```
/work/cauretc/programs/bbmap/reformat.sh in=Sequel.RunS005.001.BJE3652.subreads.bam out=stdout.fa | reformat.sh in=stdin.fa out=Sequel.RunS005.001.BJE3652.subreads.q30.fq.gz qfake=30
```
Still not working `?` everywhere. So let's use the `BJE3652.all.subreads30qfake.fq.gz` and set `racon` as `--bq -1` to disable the quality check. See the issue on the [racon github page](https://github.com/isovic/racon/issues/28)

```
bin/racon [options] <reads.fastq> <overlaps.paf> <raw_contigs.fasta> <out_consensus.fasta>

Options
  Input/Output options:
    reads               STR   Reads in FASTQ format.
    alnpath             STR   Path to a MHAP file with read-to-target overlaps.
    raw                 STR   Path to the raw contig/read sequences (output from the layout step). GFA, FASTA/FASTQ or
                              SAM formats allowed.
    out                 STR   Output consensus sequence.
        --sam            -    SAM is provided instead of MHAP. The reads file will be ignored, and seq and qual fields
                              from the SAM file will be used. [false]
        --mhap           -    Overlaps are in PAF format instead of MHAP. [false]

  Algorithm:
        --bq            FLT   Threshold for the average base quality of the input reads. If a read has average BQ <
                              specified, the read will be skipped. If value is < 0.0, filtering is disabled. [10.0]
        --use-contig-qv  -    If false, dummy QVs equal to '!' will be assigned to each contig base during window
                              consensus. Otherwise, QVs will be loaded from the contigs file if the file is in FASTQ format. [false]
    -w, --winlen        INT   Length of the window to perform POA on. [500]
        --pileup         -    Simple pileup + majority vote consensus will be performed instead of using Spoa. Much
                              faster, but less accurate. [false]
    -e, --error-rate    FLT   Maximum allowed error rate. Used for filtering faulty overlaps. [0.30]
        --ovl-margin    FLT   Fraction of the window size to overlap the windows by. [0.0]

  Control:
    -t, --threads       INT   Number of threads to use. [4]
    -b, --winbatch      INT   Size of the batch in which to process windows. After a batch is finished, consensus of
                              the windows is joined and output to file. [20000]
        --num-batches   INT   The number of batches which to process [-1]
        --start-window  INT   ID of the window to start processing from. [0]
        --erc            -    Perform error-correction instead of contig consensus. The only difference is in the type
                              of parallelization to achieve better performance. [false]

  Alignment:
    -M, --match         INT   Match score (positive value). [5]
    -X, --mismatch      INT   Mismatch penalty (negative value expected). [-4]
    -G, --gapopen       INT   Gap open penalty (negative value expected). [-8]
    -E, --gapext        INT   Gap extend penalty (negative value expected). [-6]

  Other:
    -v, --verbose       INT   Verbose level. 0 off, 1 low, 2 medium, 3 high, 4 and 5 all levels, 6-9 debug. [5]

  Other options:
    -h, --help           -    View this help. [false]
```

## Run

### minimap
The trying run failed because of `OUT_OF_MEMORY` (on 27/01/18). Previously I submitted using `--ntasks-per-node=3` and `--mem=16gb`. I think for another try I'll do `--ntasks-per-node=1` and `--mem=100gb` to see if it runs. Also since `racon` needs `.fastq` files, I'll run on the `BJE3652.all.subreads.nolengthlimit.q30.fastq.gz` created by concatenating the files created using `reformat.sh` script from `bbmap` (submitted on 29/01/18). So seemed to run nicely and ended because of time issue, so submitted again for 48h (Jan30/18). Died again because of time so submitted for 5 days on `--ntasks-per-node=4` using `--mem=100gb`, this run completed in ~2days and 15h (9/02/18).
```
/home/cauretc/project/cauretc/programs/minimap2/minimap2 -w5 -L100 -t3 /home/cauretc/scratch/pacbio_mellotrop/raw_bam/BJE3652.all.subreads.nolengthlimit.q30.fastq.gz /home/cauretc/scratch/pacbio_mellotrop/raw_bam/BJE3652.all.subreads.nolengthlimit.q30.fastq.gz > /home/cauretc/scratch/miniasm/Only_pacbio_mellotrop_minimap.paf
gzip -1 /home/cauretc/scratch/miniasm/Only_pacbio_mellotrop_minimap.paf
```
### miniasm
Not sure if it handles compressed reads file (it does for the other input file `.paf.gz`). Submitted 9/02/18 for 2 days, same memory as `minimap`, `--ntasks-per-node=1` since doesn't seem to be multithreated, using the compressed reads.

The following commands need to be run and updated if necessary whenever the `miniasm` 1st run is done.

### convert raw assembly into `.fasta`
```
gunzip -c /home/cauretc/scratch/miniasm/Only_pacbio_mellotrop_miniasm.gfa.gz | awk '{print ">"$2; print $3}' | gzip > Only_pacbio_mellotrop_miniasm_raw_assembly.fasta
```
### racon - run 1
```
/home/cauretc/project/cauretc/programs/minimap2/minimap2 Only_pacbio_mellotrop_miniasm_raw_assembly.fasta /home/cauretc/scratch/pacbio_mellotrop/raw_bam/BJE3652.all.subreads.nolengthlimit.q30.fastq.gz > Only_pacbio_mellotrop.miniasm_raw_assembly.reads_mapped.paf

/home/cauretc/project/cauretc/programs/racon/bin/racon -t 3 /home/cauretc/scratch/pacbio_mellotrop/raw_bam/BJE3652.all.subreads.nolengthlimit.q30.fastq.gz Only_pacbio_mellotrop.miniasm_raw_assembly.reads_mapped.paf Only_pacbio_mellotrop_miniasm_raw_assembly.fasta Only_pacbio_mellotrop.racon1.fasta
```
### racon - run 2
```
/home/cauretc/project/cauretc/programs/minimap2/minimap2 Only_pacbio_mellotrop.racon1.fasta /home/cauretc/scratch/pacbio_mellotrop/raw_bam/BJE3652.all.subreads.nolengthlimit.q30.fastq.gz > Only_pacbio_mellotrop.miniasm_racon1_assembly.reads_mapped.paf

/home/cauretc/project/cauretc/programs/racon/bin/racon -t 3 /home/cauretc/scratch/pacbio_mellotrop/raw_bam/BJE3652.all.subreads.nolengthlimit.q30.fastq.gz Only_pacbio_mellotrop.miniasm_racon1_assembly.reads_mapped.paf Only_pacbio_mellotrop.racon1.fasta Only_pacbio_mellotrop.racon2.fasta
```
