# [LINKS](https://github.com/bcgsc/LINKS)

## Download and installation

As suggested in a [github issue](https://github.com/bcgsc/LINKS/issues/15), needed to comment out 
```
Is /usr/lib/perl5/core_perl/CORE the correct path matching your version of PERL?

Also, keep in mind you do not need the Bloom filter functionality if you are only going to use LINKS pipelined with ARCS or ARKS:

You could simply comment out these lines (if only using LINKS with ARC/KS):

#use lib "$FindBin::Bin/./lib/bloomfilter/swig";
#use BloomFilter;
```
Couldn't make the script run without error even with having everything needed in the path and also specified using `perl -I`.

Installed on compute canada. Tried to load the modules but still doesn't work:
```
module load nixpkgs/16.09
module load links/1.8.5
```
26/02: installed Links on Iqaluk without issue and gave a try using parameters used for *P. glauca* in the example. Very poor results. Need to adjust `d` and `t`.

27/02: run quast to check on new assembly statistics

## Usage
```
Usage: ./LINKS [v1.8.5]
-f  sequences to scaffold (Multi-FASTA format, required)
-s  file-of-filenames, full path to long sequence reads or MPET pairs [see below] (Multi-FASTA/fastq format, required)
-m  MPET reads (default -m 1 = yes, default = no, optional
	! DO NOT SET IF NOT USING MPET. WHEN SET, LINKS WILL EXPECT A SPECIAL FORMAT UNDER -s
	! Paired MPET reads in their original outward orientation <- -> must be separated by ":"
	  >template_name
	  ACGACACTATGCATAAGCAGACGAGCAGCGACGCAGCACG:ATATATAGCGCACGACGCAGCACAGCAGCAGACGAC
-d  distance between k-mer pairs (ie. target distances to re-scaffold on. default -d 4000, optional)
	Multiple distances are separated by comma. eg. -d 500,1000,2000,3000
-k  k-mer value (default -k 15, optional)
-t  step of sliding window when extracting k-mer pairs from long reads
(default -t 2, optional)
	Multiple steps are separated by comma. eg. -t 10,5
-o  offset position for extracting k-mer pairs (default -o 0, optional)
-e  error (%) allowed on -d distance   e.g. -e 0.1  == distance +/- 10%
(default -e 0.1, optional)
-l  minimum number of links (k-mer pairs) to compute scaffold (default -l 5, optional) 
-a  maximum link ratio between two best contig pairs (default -a 0.3, optional)
	 *higher values lead to least accurate scaffolding*
-z  minimum contig length to consider for scaffolding (default -z 500, optional)
-b  base name for your output files (optional)
-r  Bloom filter input file for sequences supplied in -s (optional, if none provided will output to .bloom)
	 NOTE: BLOOM FILTER MUST BE DERIVED FROM THE SAME FILE SUPPLIED IN -f WITH SAME -k VALUE
	 IF YOU DO NOT SUPPLY A BLOOM FILTER, ONE WILL BE CREATED (.bloom)
-p  Bloom filter false positive rate (default -p 0.001, optional; increase to prevent memory allocation errors)
-x  Turn off Bloom filter functionality (-x 1 = yes, default = no, optional)
-v  Runs in verbose mode (-v 1 = yes, default = no, optional)
```
## Pacbio reads

We can't run mate paired reads and pacbio reads at the same time. They need to be run separately.

### 1st try - *P. glauca* parameters

The genome of the white spruce size is ~20 Gbp. It is the biggest genome from the example set. In the example set, `LINKS` was used to re-scaffold the draft genome using another draft genome (so ~1X coverage). Having a "low" coverage from the corrected (from the canu assembly preliminary step) pacbio reads, I thought the same parameters would make sense to try as a 1st step to see how the program behave. Smaller values of `-d` and `-t` should be used later considering the poor scaffolding result but it will increase the RAM usage.

#### Commands
`running_links_mello_canu.sh` to run iteratively each time on the new assembly produced: parameters used: 

- `-d` = `5000`, `7500`, `10000`, `12500`, `15000`, `20000`, `30000`, `40000`, `50000`, `60000`, `70000`, `80000`, `90000`, `100000` 

- with respectively `-t` = `200`, `200`, `150`, `150`, `150`, `75`, `50` (`50` for the other values of `-d`)

- `-k 26 -l 5 -a 0.3`

Some of the 1st commands run used (see [running_links_mello_canu.sh](https://github.com/caro46/Tetraploid_project/blob/master/some_scripts/running_links_mello_canu.sh) to have the complete commands)
```
/work/cauretc/programs/links_v1.8.5/LINKS -f /work/cauretc/2017_Mellotropicalis/pseudomolecules/allpaths/final.assembly.fasta -s mellotrop_pacbio.fof -b mellotrop_allpaths_links1 -d 5000 -t 200 -k 26 -l 5 -a 0.3 

/work/cauretc/programs/links_v1.8.5/LINKS -f /work/cauretc/2017_Mellotropicalis/scaffolding_links/mellotrop_allpaths_links1.scaffolds.fa -s mellotrop_pacbio.fof -b mellotrop_allpaths_links2 -d 7500 -t 200 -k 26 -l 5 -a 0.3
```
#### Quast
```
/work/ben/quast-4.5/quast.py /work/cauretc/2017_Mellotropicalis/pseudomolecules/allpaths/final.assembly.fasta /work/cauretc/2017_Mellotropicalis/scaffolding_links/mellotrop_allpaths_links14.scaffolds.fa.gz -o /work/cauretc/2017_Mellotropicalis/scaffolding_links/quast_results

Assembly                    final.assembly  mellotrop_allpaths_links14.scaffolds
# contigs (>= 0 bp)         320227          320128                              
# contigs (>= 1000 bp)      286034          285946                              
# contigs (>= 5000 bp)      19292           19352                               
# contigs (>= 10000 bp)     1887            1927                                
# contigs (>= 25000 bp)     6               8                                   
# contigs (>= 50000 bp)     0               0                                   
Total length (>= 0 bp)      607138367       607529644                           
Total length (>= 1000 bp)   574372221       574773954                           
Total length (>= 5000 bp)   136492195       137197052                           
Total length (>= 10000 bp)  23462309        24004275                            
Total length (>= 25000 bp)  159305          215194                              
Total length (>= 50000 bp)  0               0                                   
# contigs                   320227          320128                              
Largest contig              30521           30521                               
Total length                607138367       607529644                           
GC (%)                      38.79           38.79                               
N50                         1920            1923                                
N75                         1243            1243                                
L50                         76909           76740                               
L75                         178611          178440                              
# N's per 100 kbp           6522.06         6582.26
```
Mainly longer scaffolds, a little bit less but same N50. Hoping to have better results using mate pairs and long reads from LoRDEC.

## Mate pair reads
To create the input for `LINKS`, I used the [make_MPET_input_links.pl](https://github.com/caro46/Tetraploid_project/blob/master/some_scripts/make_MPET_input_links.pl). As specified on the `LINKS` [github page](https://github.com/bcgsc/LINKS), it 1st converts the `fastq` reads into `fasta`, then into the `LINKS` format (pair reads separated by `:`), and finally creates the `.fof` (file-of-filenames), necessary to run the program. 

Note (27/02):

On the github page it is not clear to me the actual format of `.fof` if we want to specify multiple files names... My guess was 1 name/line. We will need a try to make sure.


# [OPERA-LG](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-016-0951-y)

[Dependencies](https://sourceforge.net/p/operasf/wiki/The%20OPERA%20wiki/) (for version v2.0.6)

- samtools (only version 0.1.19 or below)

- bwa

- blasr (only version 1.3.1 or below)

I haven't been able to install `blasr` on cedar yet.

# SSPACE/SSPACE-LONGREAD

*SSPACE tracks in memory all contigs. That means that the memory usage will increase drastically with the size of your contig data set.* ([sspace_basic github](https://github.com/nsoranzo/sspace_basic)).
